src/verify-and-receive.ts

```diff
@@ -29,18 +30,29 @@ export async function verifyAndReceive(
     );
   }
 
-  let payload: EmitterWebhookEvent["payload"];
+  // The body is already an Object (e.g. GCF) and can be passed directly to the EventHandler
+  if (event.body) {
+    return state.eventHandler.receive({
+      id: event.id,
+      name: event.name,
+      payload: event.body as EmitterWebhookEvent["payload"],
+    });
+  }
+
+  const payload =
+    typeof event.payload === "string"
+      ? event.payload
+      : event.payload.toString("utf8");
+
   try {
-    payload = JSON.parse(event.payload);
+    return state.eventHandler.receive({
```

Comment:
This changes the behaviour of the try/catch

We don't want to catch errors from the `state.eventHandler.receive()` function,
We want to make sure that it is valid JSON

Reply:
done